Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,412 @@
+            {
+                resultMessage
+                    .Append(FormatStringAndCommas(peopleList))
+                    .Append(peopleMessage)
+                    .Append(FormatCurrency(poorestPersonMoney, currencySymbol))
+                    .Append(messageEnd);
+                return resultMessage.ToString();
+            }
+
+            resultMessage.Clear();
+
+            resultMessage
+                .Append(poorestPerson)
+                .Append(singlePersonMessage)
+                .Append(FormatCurrency(poorestPersonMoney, currencySymbol))
+                .Append(messageEnd);
+
+            return resultMessage.ToString();
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (!ArrayIsValid(peopleAndBalances)) return invalidMessage;
+
+            const string singlePersonMessage = " lost the most money. ";
+            var resultMessage = new StringBuilder();
+            var peopleList = new string[peopleAndBalances.Length];
+            var biggestLossList = new decimal[peopleAndBalances.Length];
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                if (TotalBalanceForSinglePerson(peopleAndBalances[i]) == CurrentBalanceForSinglePerson(peopleAndBalances[i]))
+                {
+                    return invalidMessage;
+                }
+            }
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                biggestLossList[i] = CalculateLossForSinglePerson(peopleAndBalances[i]);
+                peopleList[i] = ReturnNameForSinglePerson(peopleAndBalances[i]);
+            }
+
+            if (biggestLossList.Length > 2 && ArrayElementsAreEqual(biggestLossList))
+            {
+                resultMessage
+                    .Append(invalidMessage);
+                return resultMessage.ToString();
+            }
+
+            var biggestLossIndex = FindDecimalArrayMin(biggestLossList);
+
+            var biggestLoss = biggestLossList[biggestLossIndex];
+            var personWithBiggestLoss = peopleList[biggestLossIndex];
+
+            resultMessage.Clear();
+            resultMessage
+                .Append(personWithBiggestLoss)
+                .Append(singlePersonMessage)
+                .Append(FormatCurrency(biggestLoss, currencySymbol))
+                .Append(messageEnd);
+
+            return resultMessage.ToString();
         }
 
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (!ArrayIsValid(peopleAndBalances)) return invalidMessage;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4970157)